pages/vote.tsx

```diff
@@ -132,18 +137,18 @@ class VotePage extends React.Component<VoteProps, VoteState> {
         })
       } else {
         // if previous ordering data has been persisted then apply this and override API response ordering
-        const indicies = new Map<string, number>(JSON.parse(orderings).map((id, index) => [id, index]))
-        const preordered = sessions
-          .map(session => ({
-            index: indicies.get(session.Id) || 0,
-            session,
-          }))
-          .sort(({ index: first }, { index: second }) => first - second)
-          .map(({ session }) => session) as Session[]
+        const orderingsArray = JSON.parse(orderings)
+        const ordered = orderingsArray
+          .map(id => sessions.find(s => s.Id === id))
+          .filter(s => s)
+          .concat(sessions.filter(s => !orderingsArray.find(id => id === s.Id)))
+
+        const ids = JSON.stringify(ordered.map(({ Id }) => Id))
```

Comment:
The above may change the indices so I'm updating them too

Reply:
Cool. This approach is an improvement over the previous implementation. Previously we assigned a low value (0) if a session id did not exist in the indices retried from local storage. This would result in the order from the api response not being respected. The new solution append to the end so the order is preserved. Both behave equivalently if all sessions exist in the stored indices. As such, not an issue a user would likely encounter but worth addressing.